### DISCUSSION TITLE
send message to a user

### DISCUSSION THREAD

**Discussion by paulobreim** [DinoLeung/TeleDart#211] [Q&A] [answered]
It is possible reply a message directly to an user, not into the group?

tks

**Comment by The0xArchitect** [comment, marked as answer]
Hey, the `TeleDartMessage.reply` method is used to reply to the chat where the message is sent that is if the chat is a group chat, the bot will reply to the group itself. 

Anyway, you can use the `TeleDart.sendMessage` method to send messages to your bot users. Just pass the user's chat_id as a parameter to the `sendMessage` method.

⚠️ The only thing you must remember is that the user must have initiated the chat. (Telegram bots cannot start the chat with users).

So, probably this snippet is what you might be looking for:

```dart
teleDart.onMessage().listen((message) {
  teleDart.sendMessage(message.from?.id, "Your reply");
});
```
Once again, this will throw an error if the user hasn't chatted with the bot before.

**Reply by paulobreim** [reply]
Great !  Worked well.
Thank you.

**Reply by The0xArchitect** [reply]
Glad to hear that, happy coding :)

**Comment by paulobreim** [comment]
Well well well, bad news.
After several tests on my BOT, I found that the proven message was working, but it only works for the user who owns the Bot.
If I try to send to any other user I get the message
 403 Forbidden: bot can't initiate conversation with user

void main() async {
 final username = (await Telegram(token).getMe()).username;
  teledart = TeleDart(token, Event(username));
  print('$username ready to work.');
  teledart.start();

  teledart.onMessage(entityType: 'bot_command').listen((message) async {
    teledart.sendMessage(message.from?.id, "Your reply");
  });
}

@DinoLeung any idea?

tks

**Reply by DinoLeung** [reply]
@paulobreim as far as I know, bots are not allowed to initiate conversations with users, in other words, users have to at least message the bot once before they can receive messages from the bot.

**Reply by paulobreim** [reply]
Okay, but in this example above, the bot's response was because the user initiated a conversation.
If this user was the creator of the BOT, the answer will be private, but for other users it gives the wrong information as bot can't initiate conversation with user, because was not bot that start the conversation.

**Reply by DinoLeung** [reply]
have you tried to get the user to direct message the bot (not within a group chat) before sending user a message?

**Reply by paulobreim** [reply]
Hi Dino,
I did not try.
If there is a group with a Bot serving users, it makes no sense for the Bot to send a message to the user, in the Group itself, asking him to send a message directly to the Bot, for the Bot to answer what he asked in the Group.
I opened a call on telegram, but if they don't respond, I'll have to make a cell phone app and kill the group that has 17,000 users.

**Comment by paulobreim** [comment]
well i did a test directly in the web
https://api.telegram.org/botMYBIT/sendMessage?chat_id=USERID&text=test and really work only for the onwer  of BOT.
ok | false
error_code | 403
description | "Forbidden: bot can't initiate conversation with a user"

for me this makes no sense and limits some actions because I need to send private information to users.
This is a limitation of Telegram, I don't know if there is any way around it because it makes the app 100% unfeasible

paulo

**Reply by The0xArchitect** [reply]
I'm pretty pretty pretty sure that the user hasn't started the chat with the bot. If the user has initiated the chat, your bot will send the message to the user without an error.

The description itself is crystal clear and states `bot can't initiate conversation with a user`.

Please do double check :)

**Reply by paulobreim** [reply]
I understand what you're talking about, but look at this:
teledart.onMessage(entityType: 'bot_command').listen((message) async {
teledart.sendMessage(message.from?.id, "Your reply");});

In that teledart.onMessage, isn't the user sending a request to the bot?
If so, the line just below teledart.sendMessage could not give an error.

**Reply by DinoLeung** [reply]
@paulobreim As @HeySreelal has pointed out a couple times, the user needs to direct message the bot at least once in order to receive messages from it. There's no way around it unfortunately.

**Reply by paulobreim** [reply]
> @paulobreim As @HeySreelal has pointed out a couple times, the user needs to direct message the bot at least once in order to receive messages from it. There's no way around it unfortunately.

Thanks for the reply.
I think Telegram is making a mistake because if the user sent a command to the BOT, obviously it's already a message, so it doesn't make sense for the API to say that the user didn't initiate a message with the BOT.
The BOT is obliged to answer a particular question in the group and this only pollutes the group because other users are not interested in the answer.

Dino, you lib is fantastic.
